pkg/cloud/services/iamauth/reconcile.go

```diff
@@ -83,13 +92,110 @@ func (s *Service) ReconcileIAMAuthenticator(ctx context.Context) error {
 	return nil
 }
 
-func (s *Service) getAccountID() (string, error) {
-	input := &sts.GetCallerIdentityInput{}
+func (s *Service) getARNForRole(role string) (string, error) {
+	input := &iam.GetRoleInput{
+		RoleName: aws.String(role),
+	}
+	out, err := s.IAMClient.GetRole(input)
+	if err != nil {
+		return "", errors.Wrap(err, "unable to get role")
+	}
+	return aws.StringValue(out.Role.Arn), nil
+}
+
+func (s *Service) GetRolesForWorkers(ctx context.Context) (map[string]struct{}, error) {
```

Comment:
How about we keep the  API simple (just `ReconcileIAMAuthenticator`), and not export this?

```suggestion
func (s *Service) getRolesForWorkers(ctx context.Context) (map[string]struct{}, error) {
```

Reply:
(its resolved) but this changed after we moved the integration to the service. when the integration test was in `controlplane/eks/controllers` we needed it public